.github/workflows/llvm.yml

```
@@ -0,0 +1,109 @@
+name: LLVM unit tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      python_version:
+        required: true
+        type: string
+      ref:
+        required: false
+        type: string
+      check_run_id:
+        required: false
+        type: string
+      pr_repo:
+        required: false
+        type: string
+  push:
+    branches: [devel, main, compile-with-clang]
+
+env:
+  COMMIT: ${{ inputs.ref || github.event.ref }}
+  PEM: ${{ secrets.BOT_PEM }}
+  GITHUB_RUN_ID: ${{ github.run_id }}
+  GITHUB_CHECK_RUN_ID: ${{ inputs.check_run_id }}
+  PR_REPO: ${{ inputs.pr_repo || github.repository }}
+  PYCCEL_DEFAULT_COMPILER: 'LLVM'
+
+jobs:
+  Python_version_picker:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'push' || github.repository == 'pyccel/pyccel'
+    outputs:
+      python_version: ${{ steps.set-python_version.outputs.python_version }}
+    steps:
+    - if: github.event_name == 'push'
+      run: |
+        echo "version=3.11" >> $GITHUB_ENV
+      shell: bash
+    - id: dispatch-matrix
+      if: github.event_name == 'workflow_dispatch'
+      run: |
+        echo "version=${{ inputs.python_version }}" >> $GITHUB_ENV
+      shell: bash
+    - id: set-python_version
+      run: |
+        echo "python_version=$version" >> $GITHUB_OUTPUT
+      shell: bash
+
+  LLVM:
+
+    runs-on: ubuntu-latest
+    name: Unit tests
+    needs: Python_version_picker
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ env.COMMIT }}
+          repository: ${{ env.PR_REPO }}
+          submodules: true
+      - name: Set up Python ${{ needs.Python_version_picker.outputs.python_version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ needs.Python_version_picker.outputs.python_version }}
+      - name: "Setup"
+        if: github.event_name != 'push'
+        id: token
+        run: |
+          pip install jwt requests
+          python ci_tools/setup_check_run.py LLVM
+      - name: Install dependencies
+        uses: ./.github/actions/llvm_install
+      - name: Install Pyccel with tests
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install .[test]
+        shell: bash
+      - name: Coverage install
+        uses: ./.github/actions/coverage_install
+      - name: Fortran/C tests with pytest
+        id: f_c_pytest
+        timeout-minutes: 60
+        uses: ./.github/actions/pytest_run
+      - name: Python tests with pytest
+        id: python_pytest
+        timeout-minutes: 20
+        uses: ./.github/actions/pytest_run_python
+      - name: Parallel tests with pytest
+        id: parallel
+        timeout-minutes: 20
+        uses: ./.github/actions/pytest_parallel
+      - name: Test with valgrind for memory leaks
+        id: valgrind
+        uses: ./.github/actions/valgrind_run
```

It would be nice to merge what you have. So let's allow this failure for now:
```suggestion
      - name: Test with valgrind for memory leaks
        id: valgrind
        continue-on-error: true
        uses: ./.github/actions/valgrind_run
```
and open a dedicated issue

---

Oh, you want to merge it for v2.0?

---

Done in [134ab22](https://github.com/pyccel/pyccel/pull/2261/commits/134ab224b57562bf83f98b0d030b3f0e57ac4b0e)